lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -1125,6 +1125,16 @@ LogicalResult InferResetsPass::updateReset(ResetNetwork net, ResetKind kind) {
       uop.replaceAllUsesWith(uop.getInput());
       LLVM_DEBUG(llvm::dbgs() << "- Inferred " << uop << "\n");
       uop.erase();
+    } else if (auto constCastOp = dyn_cast<ConstCastOp>(wop);
+               constCastOp &&
+               !areTypesConstCastable(constCastOp.getResult().getType(),
+                                      constCastOp.getInput().getType())) {
+      // Update const cast results
+      auto result = constCastOp.getResult();
+      result.setType(constCastOp.getInput().getType().getConstType(false));
+      for (auto *user : result.getUsers())
+        worklist.insert(user);
+      LLVM_DEBUG(llvm::dbgs() << "- Inferred " << constCastOp << "\n");
```

Comment:
Can you move the const castable check in to the body and add a comment explaining what is going on here?

Reply:
Updated